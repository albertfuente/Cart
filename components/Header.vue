<template>
  <div class="header-component">
    <div class="header-component-links-container">
      <div
        class="header-component-link"
        v-for="(l, index) in headerLinks"
        v-bind:key="l.label + index"
      >
        <a class="header-component-anchor" :href="l.link">
          {{ l.label }}
        </a>
      </div>
    </div>
    <img src="../static/logo.svg" class="header-component-logo" alt="mapit-logo"/>
    <div>MIS MOTOS</div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      headerLinks: [
        {
          label: "Inicio",
          link: "/",
        },
        {
          label: "Contacto",
          link: "/contact",
        },
        {
          label: "Nuestro Blog",
          link: "/blog",
        },
      ],
    };
  },
};
</script>
<style scoped>
.header-component {
  border: 1px solid #ff8000;
  margin: 10px;
  padding: 10px;
  font-size: 20px;
  font-weight: bold;
}
.header-component-links-container {
  display: flex;
  flex-direction: row;
  justify-content: end;
}
.header-component-link {
  margin: 0px 20px;
}
.header-component-anchor {
  text-decoration: none;
  color: #ff8000;
  font-size: 16px;
}
</style>