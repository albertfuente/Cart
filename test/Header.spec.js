import { shallowMount, mount } from '@vue/test-utils'
import Header from './../components/Header.vue'

describe('Header.vue/Header', () => {
  it('mounts the component and creates a snapshot inside the __snapshots__ folder', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper.html()).toMatchSnapshot()
  })


  it('does not display the Header if items are empty ', () => {
    const items = []
    const wrapper = mount(Header, { propsData: { items } })

    expect(wrapper.find('.header-component-items-container').exists()).toBe(false)
  })
})
