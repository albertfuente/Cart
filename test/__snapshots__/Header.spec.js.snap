// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue/Header mounts the component and creates a snapshot inside the __snapshots__ folder 1`] = `
"<div class=\\"header-component\\">
  <div class=\\"header-component-links-container\\">
    <div class=\\"header-component-link\\"><a href=\\"/\\" class=\\"header-component-anchor\\">
        Inicio
      </a></div>
    <div class=\\"header-component-link\\"><a href=\\"/contact\\" class=\\"header-component-anchor\\">
        Contacto
      </a></div>
    <div class=\\"header-component-link\\"><a href=\\"/blog\\" class=\\"header-component-anchor\\">
        Nuestro Blog
      </a></div>
  </div> <img src=\\"../static/logo.svg\\" alt=\\"mapit-logo\\" class=\\"header-component-logo\\">
  <div>MIS MOTOS</div>
</div>"
`;
